src/osd/OSD.cc

```diff
@@ -4287,6 +4304,51 @@ int OSD::shutdown()
     cct->_conf.apply_changes(nullptr);
   }
 
+  if (cct->_conf->osd_fast_shutdown) {
+    // first, stop new task from being taken from op_shardedwq
+    op_shardedwq.stop_for_fast_shutdown();
+
+    // do we still need to drain ???
+    //op_shardedwq.drain();
+
+    utime_t  start_time_timer = ceph_clock_now();
+    // TBD: Is this too early???
```

Comment:
I think this should be ok at this point. May want to stop tick_timer_without_osd_lock as well - it could have generated more work (which is being discarded, but it's still not necessary)